### DISCUSSION TITLE
Обход блокировок ovpn/wireguard возможен или только http/s ?

### DISCUSSION THREAD

**Discussion by C0wS0ft** [bol-van/zapret#209] [Q&A]
Готовиться надо к худшему, хотя уже сейчас билайн мобайл постоянно блокирует.

**Comment by spvkgn** [comment]
Тоже интересует этот вопрос после недавних мер РКН против WG. Форк WG от AmneziaVPN пока что способен этому противостоять с помощью "мусора" при хендшейке (параметры Jc, Jmin, Jmax). Возможно ли настроить аналогичный функционал в zapret?

**Comment by bol-van** [comment]
Возможно. nfqws даже распознает wireguard handshake, ему не нужен desync any protocol
Но удобного провода через скрипты нет
Или руками, или через кастом скрипт, или по-извращенному, но простому, варианту включаем QUIC, добавляем порт WG в QUIC_PORTS и вешаем стратегию fake на QUIC. Возможно, с repeats=6 или 12
Уже год этим пользуюсь на теле2

**Reply by spvkgn** [reply]
> или по-извращенному, но простому, варианту включаем QUIC, добавляем порт WG в QUIC_PORTS и вешаем стратегию fake на QUIC. Возможно, с repeats=6 или 12

Да, вполне рабочий вариант, проверил с WARP. Но наверное лучше в custom mode запускать, пока не разобрался с этим.

**Reply by Alexey71** [reply]
> Возможно. nfqws даже распознает wireguard handshake, ему не нужен desync any protocol Но удобного провода через скрипты нет Или руками, или через кастом скрипт, или по-извращенному, но простому, варианту включаем QUIC, добавляем порт WG в QUIC_PORTS и вешаем стратегию fake на QUIC. Возможно, с repeats=6 или 12 Уже год этим пользуюсь на теле2

В новом конфиге получается так прописывать? Порт нужен который в конечной точке. или порт интерфейса?
```
NFQWS_PORTS_UDP=56789
NFQWS_OPT="--filter-udp=56789 --dpi-desync=fake --dpi-desync-repeats=12"
```

**Reply by spvkgn** [reply]
> В новом конфиге получается так прописывать? Порт нужен который в конечной точке. или порт интерфейса?
> ```
> NFQWS_PORTS_UDP=56789
> NFQWS_OPT="--filter-udp=56789 --dpi-desync=fake --dpi-desync-repeats=12"
> ```

Проще через скрипт в custom.d
Порты серверов WG

**Reply by Alexey71** [reply]
> > В новом конфиге получается так прописывать? Порт нужен который в конечной точке. или порт интерфейса?
> > ```
> > NFQWS_PORTS_UDP=56789
> > NFQWS_OPT="--filter-udp=56789 --dpi-desync=fake --dpi-desync-repeats=12"
> > ```
> 
> Проще через скрипт в custom.d Порты серверов WG

Можете скрипт для примера скинуть?

**Reply by bol-van** [reply]
Если фильтр нужен только по портам, то не надо никаких custom.
Но если хотите еще пришить туда IP назначения, то надо.

**Reply by Alexey71** [reply]
Не получилось у меня завести warp на wireguard через zapret.
```
2024-10-29 12:13:43.362: [TUN] [2-OpenWrt] peer(bmXO…fgyo) - Sending handshake initiation
2024-10-29 12:13:48.373: [TUN] [2-OpenWrt] peer(bmXO…fgyo) - Sending handshake initiation
2024-10-29 12:13:53.670: [TUN] [2-OpenWrt] peer(bmXO…fgyo) - Handshake did not complete after 5 seconds, retrying (try 2)
2024-10-29 12:13:53.670: [TUN] [2-OpenWrt] peer(bmXO…fgyo) - Sending handshake initiation
2024-10-29 12:13:58.939: [TUN] [2-OpenWrt] peer(bmXO…fgyo) - Handshake did not complete after 5 seconds, retrying (try 2)
2024-10-29 12:13:58.939: [TUN] [2-OpenWrt] peer(bmXO…fgyo) - Sending handshake initiation
```

**Reply by bol-van** [reply]
Если убедиться в логе nfqws, что есть реакция на пакеты и выполняется дурение, то вероятно не работает фейк.
Может по IP забанено, может stateless. Если это известный сервис VPN, а не своя VPSка, то такой вариант вполне возможен

**Reply by spvkgn** [reply]
> Не получилось у меня завести warp на wireguard через zapret.

У меня просто с fake и повторами тоже уже не работает, по кр. мере на 162.159.192.x и 193.x адреса. Другое решение нужно. Хотя, если там ответ сервера смотрят, то не факт что поможет. А так с другими сервисами пока работает.

**Reply by spvkgn** [reply]
> У меня просто с fake и повторами тоже уже не работает, по кр. мере на 162.159.192.x и 193.x адреса.

Что-то там разморозилось - сегодня все нормально фейком пробивается

**Reply by Alexey71** [reply]
> > У меня просто с fake и повторами тоже уже не работает, по кр. мере на 162.159.192.x и 193.x адреса.
> 
> Что-то там разморозилось - сегодня все нормально фейком пробивается

Вы тоже прописываете как я тут?
https://github.com/bol-van/zapret/discussions/209#discussioncomment-11082666

@bol-van А для чего нужны wireguard_initiation.bin и wireguard_response.bin в папке zapret/files/fake ?

**Reply by spvkgn** [reply]
> Вы тоже прописываете как я тут?

Примерно то же самое:  [50-wireguard.zip](https://github.com/user-attachments/files/17596268/50-wireguard.zip)

**Reply by v-ethereal** [reply]
>Уже год этим пользуюсь на теле2
Друг, не раскрывай про себя ничего. Совсем. Никогда.

**Reply by recrumptor** [reply]
В виду последних событий, даже Амнезия с Jc/Jmin/Jmax уже не может подключиться по wireguard к большинству серверов варп. Есть смысл искать рабочую стратегию zapret?

**Comment by wowik98** [comment]
настроил на keenetic 
config такой 
NFQWS_PORTS_TCP=80,443
NFQWS_PORTS_UDP=443, (порты Wireguard )
NFQWS_OPT="
--filter-tcp=80 --dpi-desync=fake --dpi-desync-repeats=10 --dpi-desync-ttl=4 --dpi-desync-fooling=md5sig --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin --new 
--filter-tcp=443 --dpi-desync=fake --dpi-desync-repeats=10 --dpi-desync-ttl=4 --dpi-desync-fooling=md5sig --dpi-desync-fake-tls=/opt/zapret/files/fake/tls_clienthello_www_google_com.bin --new 
--filter-udp=443 --dpi-desync=fake --dpi-desync-repeats=12 --dpi-desync-any-protocol --new
--filter-l7=wireguard --dpi-desync=fake --dpi-desync-repeats=12 --dpi-desync-any-protocol --new

работает только на роуторе а на пк c wind если запустить Wireguard  один и тот же конфиг  получаю
Sending handshake initiation
Routine: encryption worker - stopped
на Android  is beyond type entryCount(219)
может что не так с config  подскажите

**Reply by NOFEXtreme** [reply]
`--dpi-desync-any-protocol` для wireguard не нужен. У меня с wg больше 2 repeats не работает. Можно попробовать использовать `--dpi-desync-fake-wireguard=<filename>`
У меня для wg на уфанет вот так:
```
--filter-udp=51820 --filter-l7=wireguard --dpi-desync=fake --dpi-desync-fake-wireguard=$ZAPRET_BASE/files/fake/tls_clienthello_www_google_com.bin
```

**Reply by bol-van** [reply]
`--filter-l7=wireguard --dpi-desync=fake --dpi-desync-repeats=12 --dpi-desync-any-protocol --new`

Вот эта конструкция излишня.
Сначала ищем wireguard, зачем почему-то any protocol.
Если он может распознать wireguard, зачем долбить каждый пакет, кроме handshake initiation по 12 раз
Не дай бог это на винду перенесут как есть и будет 12 кратный срач без cutoff

**Reply by wowik98** [reply]
> `--filter-l7=wireguard --dpi-desync=fake --dpi-desync-repeats=12 --dpi-desync-any-protocol --new`
> 
> Вот эта конструкция излишня. Сначала ищем wireguard, зачем почему-то any protocol. Если он может распознать wireguard, зачем долбить каждый пакет, кроме handshake initiation по 12 раз Не дай бог это на винду перенесут как есть и будет 12 кратный срач без cutoff

у вы не помогло
получено 92б
ну хоть на роуторе работает

**Comment by Artem-VV** [comment]
С такими аргументами получилось добиться небольшой стабильности. WG подключается на 443 по домену.
`--filter-udp=443 --filter-l7=wireguard --dpi-desync=fake --dpi-desync-repeats=6 --dpi-desync-fake-wireguard="/usr/local/etc/dvtws/bin/quic_initial_www_google_com.bin"`
Есть какие-то рекомендации?

**Reply by Artem-VV** [reply]
Данный метод перестал работать после отвала интернета в РФ. У кого-то есть рабочие стратегии или идеи их создания?

**Reply by spvkgn** [reply]
> Данный метод перестал работать после отвала интернета в РФ. У кого-то есть рабочие стратегии или идеи их создания?

Клиент AmneziaWG тоже никак не помогает?

**Reply by Artem-VV** [reply]
> > Данный метод перестал работать после отвала интернета в РФ. У кого-то есть рабочие стратегии или идеи их создания?
> 
> Клиент AmneziaWG тоже никак не помогает?

Амнезию нет возможности использовать ввиду того, что в качестве роутера у меня opnsense (freebsd). И в прицнипе я нашел проблему, а точнее две: для домена, через который я подключаюсь, не был настроен tls 1.3, а также у хостера были проблемы в сети. Сейчас все стабильно работает в таком исполнении:
--filter-udp=443 --hostlist="list-general.txt" --dpi-desync=fake --dpi-desync-repeats=6 --dpi-desync-fake-quic="/usr/local/etc/dvtws/bin/quic_initial_vk_com.bin"

Домен вайргарда был дописан в хостлист просто, чтобы не городить дополнительные политики.

Касаемо амнезии: конечно есть возможность родить виртуалку конкретно для амнезии и использовать ее как шлюз где нужно, но заниматься этим мне попросту лень :)

**Reply by spvkgn** [reply]
> Амнезию нет возможности использовать ввиду того, что в качестве роутера у меня opnsense (freebsd).

Клиент же можно за роутером запустить для проверки, с пк или андроида

**Reply by Artem-VV** [reply]
> > Амнезию нет возможности использовать ввиду того, что в качестве роутера у меня opnsense (freebsd).
> 
> Клиент же можно за роутером запустить для проверки, с пк или андроида

Я понимаю, но мне нужен впн на всю сеть ввиду большого количества устройств. На телефон/пк у меня xray.

**Reply by Lintech-1** [reply]
> > > Данный метод перестал работать после отвала интернета в РФ. У кого-то есть рабочие стратегии или идеи их создания?
> > 
> > 
> > Клиент AmneziaWG тоже никак не помогает?
> 
> Амнезию нет возможности использовать ввиду того, что в качестве роутера у меня opnsense (freebsd). И в прицнипе я нашел проблему, а точнее две: для домена, через который я подключаюсь, не был настроен tls 1.3, а также у хостера были проблемы в сети. Сейчас все стабильно работает в таком исполнении: --filter-udp=443 --hostlist="list-general.txt" --dpi-desync=fake --dpi-desync-repeats=6 --dpi-desync-fake-quic="/usr/local/etc/dvtws/bin/quic_initial_vk_com.bin"
> 
> Домен вайргарда был дописан в хостлист просто, чтобы не городить дополнительные политики.
> 
> Касаемо амнезии: конечно есть возможность родить виртуалку конкретно для амнезии и использовать ее как шлюз где нужно, но заниматься этим мне попросту лень :)

А какие вы домены прописывали? можете поделиться?

**Reply by bol-van** [reply]
амнезнию с параметрами Jc/Jmin/Jmax обычно можно заменить на wireguard+фейки от zapret. возможно, придется поиграться репитами и собственными фейк пейлоадами разной длины.
amnezia делает тоже самое, но гененрит рандом содержание и длину каждый раз
